src/datachain/query/dataset.py

```diff
@@ -982,6 +982,9 @@ def apply(
 
         columns1, columns2 = _order_columns(q1.columns, q2.columns)
 
+        columns1 = _remap_sys_id_dense(columns1, branch_offset=0)
+        columns2 = _remap_sys_id_dense(columns2, branch_offset=1)
```

Comment:
We should also remap `sys__random` field. After `union` there might (will?) be rows with similar `sys__random` field, and we want (?) `sys__random` to be different across all table.

Reply:
> The idea (from very beginning, I guess, I am not sure, we do have this field from the very beginning) was to always have this sys__rand field in the DB, just in case we need any random filtering/order/etc

omg, this way way too heavy (and probably doesn't work anyways in many cases) :(

Reply:
okay, I guess I will add it, but it is bad design (too heavy if we just use it for split)